modules/core/src/lifecycle/prop-types.ts

```diff
@@ -192,8 +192,22 @@ const TYPE_DEFINITIONS = {
   },
   data: {
     transform: (value, propType: DataPropType, component) => {
+      if (!value) {
+        return value;
+      }
       const {dataTransform} = component.props;
-      return dataTransform && value ? dataTransform(value) : value;
+      if (dataTransform) {
+        return dataTransform(value);
```

Comment:
Won't apps that use `dataTransform` break when they encounter the new format? We should at least document in the upgrade guide